File: impl/app/android/app/src/main/java/com/ak/contexta/domain/repository/StatsRepository.kt
package com.ak.contexta.domain.repository

import com.ak.contexta.domain.model.DailyStats
import kotlinx.coroutines.flow.Flow

interface StatsRepository {
    fun observeStats(): Flow<DailyStats?>

    suspend fun getStats(): DailyStats?

    /** Record reading activity for today. */
    suspend fun recordReadingActivity(secondsSpent: Int = 0)

    /** Record adding a word to vocabulary. */
    suspend fun recordWordAdded()
}

File: impl/app/flutter/lib/domain/repository/stats_repository.dart
import '../model/daily_stats.dart';

/// 统计仓储接口（对齐 Kotlin StatsRepository.kt）。
abstract interface class StatsRepository {
  /// 观察今日学习统计（无记录时发射 null）。
  Stream<DailyStats?> observeStats();

  Future<DailyStats?> getStats();

  /// 记录今天的阅读活动（+1 篇文章 + 秒数）。
  Future<void> recordReadingActivity({int secondsSpent = 0});

  /// 记录今天加入生词本一个词。
  Future<void> recordWordAdded();
}
